backend/inspirehep/pidstore/providers/cnum.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from datetime import datetime
+
+import structlog
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+from invenio_pidstore.providers.base import BaseProvider
+
+LOGGER = structlog.getLogger()
+
+
+class CNUMIdentifier(object):
+    def next(self, data):
+        """Generate a CNUM identifier from a conference record.
+
+        Args:
+            data (dict): the record metadata
+
+        Returns:
+            str: a conference CNUM identifier in the form CYY-MM-DD[.X]
+        """
+        opening_date = data.get("opening_date")
+        if not opening_date:
+            return
+
+        adjusted_date = datetime.strptime(opening_date, "%Y-%m-%d").strftime("%y-%m-%d")
+        cnum = f"C{adjusted_date}"
+        cnums_count = PersistentIdentifier.query.filter(
+            PersistentIdentifier.pid_value.like(f"{cnum}%")
```

Comment:
I think you also want to set the `pid_type` here to
1. ensure you don't match something that looks like a CNUM but is not
2. use the index

Reply:
yeah why not :+1: